netket/jax/_utils_dtype.py

```diff
@@ -83,3 +85,15 @@ def maybe_promote_to_complex(*types):
             return dtype_complex(main_typ)
     else:
         return main_typ
+
+
+def canonicalize_dtypes(*values, dtype=None):
+    """
+    If `dtype` is None, determine it by promoting `values` in JAX.
+    The returned dtype is always cast to x32 if x64 is disabled.
```

Comment:
```
Return the canonicalised result dtype of an operation combining several values, with
a possible default dtype.

Equivalent to 
```python
    if dtype is None:
        dtype = jnp.result_type(*[_dtype(x) for x in values])
    return jax.dtypes.canonicalize_dtype(dtype)
```

Args:
    *values: all values to combine. Ignored if dtype is not None
    dtype: default value overriding values.

Reply:
please update the docstring

Reply:
no ok, did it myself